cmd_migrate_db.go

```diff
@@ -0,0 +1,650 @@
+		if err == nil {
+			logger.Infof("Skipping DB with prefix %s because the "+
+				"source DB was marked with a tombstone which "+
+				"means it was already migrated successfully. "+
+				"Tombstone reads: %s", prefix, marker)
+
+			continue
+		}
+		if err != channeldb.ErrMarkerNotPresent {
+			return err
+		}
+
+		// Check that the source DB has had all its schema migrations
+		// applied before we migrate any of its data. Currently only
+		// migration of the channel.db and the watchtower.db exist.
+		//
+		// TODO(ziggie): Use the tagged LND 19 version here.
+
+		// Check channel.db migration.
+		if prefix == lncfg.NSChannelDB {
+			logger.Info("Checking DB version of source DB " +
+				"(channel.db)")
+
+			err := checkChannelDBMigrationsApplied(srcDb)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Check watchtower client DB migrations.
+		if prefix == lncfg.NSTowerClientDB {
+			logger.Info("Checking DB version of source DB " +
+				"(wtclient.db)")
+
+			err := checkWTClientDBMigrationsApplied(srcDb)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Also make sure that the destination DB hasn't been marked as
+		// successfully having been the target of a migration. We only
+		// mark a destination DB as successfully migrated at the end of
+		// a successful and complete migration.
+		logger.Info("Checking if migration was already applied to " +
+			"destination DB")
+		marker, err = checkMarkerPresent(destDb, alreadyMigratedKey)
+		if err == nil {
+			logger.Infof("Skipping DB with prefix %s because the "+
+				"destination DB was marked as already having "+
+				"been the target of a successful migration. "+
+				"Tag reads: %s", prefix, marker)
+
+			continue
+		}
+		if err != channeldb.ErrMarkerNotPresent {
+			return err
+		}
+
+		// Configure and run migration.
+		cfg := migratekvdb.Config{
+			Logger:            logger.SubSystem("MIGKV"),
+			ForceNewMigration: x.ForceNewMigration,
+			ChunkSize:         x.ChunkSize,
+		}
+
+		migrator, err := migratekvdb.New(cfg)
+		if err != nil {
+			return err
+		}
+
+		err = migrator.Migrate(context.Background(), srcDb, destDb)
+		if err != nil {
+			return err
+		}
+		logger.Info("Migration completed")
+
+		// We migrated the DB successfully, now we verify the migration.
+		err = migrator.VerifyMigration(context.Background(), destDb)
+		if err != nil {
+			return err
+		}
+
+		logger.Info("Verification completed")
+
+		// Migrate wallet created marker. This is done after the
+		// migration to ensure the verification of the migration
+		// succeeds.
+		if prefix == lncfg.NSWalletDB && !x.Dest.isBoltDB() {
+			err := createWalletMarker(destDb, logger)
+			if err != nil {
+				return err
+			}
+		}
+
+		// If we get here, we've successfully migrated the DB and can
+		// now set the tombstone marker on the source database and the
+		// already migrated marker on the target database.
+		if err := addMarker(srcDb, channeldb.TombstoneKey); err != nil {
```

Comment:
Totally non-blocking, but something that would be useful for the user (and us when troubleshooting with users) is if we would also create an empty `<dbname>.db.migrated-to-sqlite-yyyy-mm-dd` file if the source database was a `bbolt` one. That way just from `ls -lah` you'd see: ah, I migrated the DB over to sqlite, so I can probably delete the bolt DB now.